### ISSUE TITLE
mysql_database :query fail as of 4.0.9

### ISSUE THREAD

**Issue by krtyyy** [chef-cookbooks/database#170]
We've got some pretty standard MySQL commands (straight from your examples) that work just fine in 4.0.8 and earlier. As soon as we pick up 4.0.9, we get the following on all of our mysql_database :query actions. 

Are we missing something? Is there a workaround other than pinning database to 4.0.8? 

Some additional context (if it matters; we can use these with database 4.0.8 and it all works as before):
OS: RHEL 6.6
Some other cookbooks:
mysql 6.0.31
mysql2_chef_gem 1.0.2
postgresql  3.4.20

Thanks,

Ken

`````` NoMethodError
    -------------
    undefined method `close' for nil:NilClass

    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/database/libraries/provider_database_mysql.rb:153:in `close_query_client'
    /var/chef/cache/cookbooks/database/libraries/provider_database_mysql.rb:96:in `ensure in block in <class:Mysql>'
    /var/chef/cache/cookbooks/database/libraries/provider_database_mysql.rb:96:in `block in <class:Mysql>'

    Resource Declaration:
    ---------------------    
    122: mysql_database 'flush repl privileges' do
    123:   connection mysql_connection_info
    124:   sql 'flush privileges'
    125:   action :query
    126: end```
``````

**Comment by krtyyy** [comment]
I forgot to mention that other MySQL actions (like mysql_database_user :grant, and mysql_database :create) are working just fine. It's just :query where we're seeing the problem.

**Comment by thorstenk** [comment]
see #173

tl;dr you have to set the `database_name` parameter

**Comment by krtyyy** [comment]
Will do, thanks!

**Comment by tas50** [comment]
At this time I'm going to close this issue out as we've deprecated this cookbook. The original intent was to abstract database implementation details away from the end user. In hindsight this was a mistake as it is difficult / impossible to make a generic interface for databases that works across all systems. Instead we plan to move the individual providers into their respective cookbooks (mysql, postgresql, sqlite, sql_server) where they can fully utilize the feature set of the underlying DB systems.

**Comment by krtyyy** [comment]
This is an automatically generated message.

krt@yahoo-inc.com is no longer with Yahoo! Inc.

Your message will not be forwarded.

If you have a sales inquiry, please email yahoosales@yahoo-inc.com and someone will follow up with you shortly.

If you require assistance with a legal matter, please send a message to legal-notices@yahoo-inc.com

Thank you!
